app/app_main.c

```diff
@@ -3399,6 +3449,218 @@ static ACVP_RESULT app_dsa_handler(ACVP_TEST_CASE *test_case)
     return ACVP_SUCCESS;
 }
 
+static EC_POINT *make_peer(EC_GROUP *group, BIGNUM *x, BIGNUM *y)
+{
+    EC_POINT *peer;
+    int rv;
+    BN_CTX *c;
+
+    peer = EC_POINT_new(group);
+    if (!peer) {
+        printf("EC_POINT_new failed\n");
+        return NULL;
+    }
+    c = BN_CTX_new();
+    if (!c) {
+        printf("BN_CTX_new failed\n");
+        return NULL;
+    }
+    if (EC_METHOD_get_field_type(EC_GROUP_method_of(group))
+        == NID_X9_62_prime_field)
+        rv = EC_POINT_set_affine_coordinates_GFp(group, peer, x, y, c);
+    else
+#ifdef OPENSSL_NO_EC2M
+    {
+        printf("ERROR: GF2m not supported\n");
+        return NULL;
+    }
+#else
+    rv = EC_POINT_set_affine_coordinates_GF2m(group, peer, x, y, c);
+#endif
+
+    BN_CTX_free(c);
+    if (rv) {
+        return peer;
+    }
+    EC_POINT_free(peer);
+    return NULL;
+}
+
+static int ec_print_key(ACVP_KAS_ECC_TC *tc, EC_KEY *key, int add_e, int exout)
+{
+    const EC_POINT *pt;
+    const EC_GROUP *grp;
+    const EC_METHOD *meth;
+    int rv = 0;
+    BIGNUM *tx, *ty;
+    const BIGNUM *d = NULL;
+    BN_CTX *ctx;
+
+    ctx = BN_CTX_new();
+    if (!ctx) {
+        printf("BN_CTX_new failed\n");
+        return 0;
+    }
+    tx = BN_CTX_get(ctx);
+    ty = BN_CTX_get(ctx);
+    if (!tx || !ty) {
+        printf("BN_CTX_get failed\n");
+        return 0;
+    }
+    grp = EC_KEY_get0_group(key);
+    pt = EC_KEY_get0_public_key(key);
+    if (exout)
+        d = EC_KEY_get0_private_key(key);
+    meth = EC_GROUP_method_of(grp);
+    if (EC_METHOD_get_field_type(meth) == NID_X9_62_prime_field)
+        rv = EC_POINT_get_affine_coordinates_GFp(grp, pt, tx, ty, ctx);
+    else
+#ifdef OPENSSL_NO_EC2M
+    {
+        printf("ERROR: GF2m not supported\n");
+        return 0;
+    }
+#else
+    rv = EC_POINT_get_affine_coordinates_GF2m(grp, pt, tx, ty, ctx);
+#endif
+
+
+    if (add_e) {
+        tc->pix = BN_bn2hex(tx);
+        tc->piy = BN_bn2hex(ty);
+        tc->pixlen = strlen(tc->pix);
+        tc->piylen = strlen(tc->piy);
+        if (d) {
+            tc->d = BN_bn2hex(d);
+            tc->dlen = strlen(tc->d);
```

Comment:
Strnlen instead of strlen? (general comment)

Reply:
got it.